Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/cut_with_plane.h

```diff
@@ -0,0 +1,454 @@
+#include <CGAL/Polygon_mesh_processing/self_intersections.h>
+#endif
+
+namespace CGAL {
+namespace Polygon_mesh_processing {
+
+#ifndef DOXYGEN_RUNNING
+template <class PolygonMesh>
+struct Default_cut_visitor
+{
+  /// called before splitting an edge
+  void before_edge_split(typename boost::graph_traits<PolygonMesh>::halfedge_descriptor, PolygonMesh&) {}
+  /// called after an edge split, `h` = the new halfedge pointing to the new vertex
+  void edge_split(typename boost::graph_traits<PolygonMesh>::halfedge_descriptor, PolygonMesh&) {}
+  /// gives access to all the vertex that are on the cutting plane
+  void vertices_on_cut(const std::vector<typename boost::graph_traits<PolygonMesh>::vertex_descriptor>&, PolygonMesh&){}
+};
+
+// TODO: doc me or hide me in the np
+template <class Kernel>
+struct Orthogonal_cut_plane_traits
+{
+  using FT = typename Kernel::FT;
+  using Plane_3 = std::pair<int, FT>;
+  using Point_3 = typename Kernel::Point_3;
+
+  struct Oriented_side_3
+  {
+    Oriented_side operator()(const Plane_3& plane, const Point_3& p)  const
+    {
+      if (p[plane.first]==plane.second) return ON_ORIENTED_BOUNDARY;
+      return p[plane.first]<plane.second ? ON_NEGATIVE_SIDE : ON_POSITIVE_SIDE;
+    }
+  };
+
+  // TODO here we should integrate the epsilon to reuse existing points (moving them on the grid as an option?)
+  struct Construct_plane_line_intersection_point_3
+  {
+    Point_3 operator()(const Plane_3& plane, const Point_3& p, const Point_3& q)
+    {
+      //TODO: divide by largest value?
+      FT alpha = (plane.second - q[plane.first]) / (p[plane.first] - q[plane.first]);
+      std::array<FT,3> coords;
+      for (int i=0;i<3;++i)
+      {
+        if (i==plane.first)
+          coords[i] = plane.second;
+        else
+          coords[i] = (p[i]==q[i])?p[i]:p[i]*alpha +(1-alpha)*q[i];
+      }
+      return Point_3(coords[0], coords[1], coords[2]);
+    }
+  };
+
+  Oriented_side_3 oriented_side_3_object() const
+  {
+    return Oriented_side_3();
+  }
+
+  Construct_plane_line_intersection_point_3 construct_plane_line_intersection_point_3_object() const
+  {
+    return Construct_plane_line_intersection_point_3();
+  }
+
+#ifndef CGAL_PLANE_CLIP_DO_NOT_USE_BOX_INTERSECTION_D
+// for does self-intersect
+  using Segment_3 = typename Kernel::Segment_3;
+  using Triangle_3 = typename Kernel::Triangle_3;
+  using Construct_segment_3 = typename Kernel::Construct_segment_3;
+  using Construct_triangle_3 =typename  Kernel::Construct_triangle_3;
+  using Do_intersect_3 = typename Kernel::Do_intersect_3;
+  Construct_segment_3 construct_segment_3_object() const { return Construct_segment_3(); }
+  Construct_triangle_3 construct_triangle_3_object() const { return Construct_triangle_3(); }
+  Do_intersect_3 do_intersect_3_object() const { return Do_intersect_3(); }
+#endif
+};
+#endif
+
+/*!
+ *  \ingroup PMP_corefinement_grp
+ *
+ *  refines `pm` by inserting the intersection points of `plane` with its edges and faces.
```

Comment:
what's the interesting point of the plane with a face?

Reply:
I don't get the question